src/Polly.Core.Tests/Registry/StrategyId.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+
+namespace Polly.Core.Tests.Registry;
+
+public record StrategyId(Type Type, string BuilderName, string InstanceName = "")
```

Comment:
@geeknoid We can consider moving this to the core API as we would propably have to copy-paste this piece to our codebase. This struct allows:

- Isolation of strategies by the type of the result they are handling. 
- Multi-instance support. 

cc @martincostello